src/main/java/org/mockito/internal/creation/instance/ConstructorInstantiator.java

```diff
@@ -47,18 +51,11 @@ public ConstructorInstantiator(Object outerClassInstance) {
         return (T) constructor.newInstance(params);
     }
 
-    private static <T> InstantiationException paramsException(Class<T> cls, Exception cause) {
-        return new InstantiationException(
-                join("Unable to create instance of '" + cls.getSimpleName() + "'.",
-                     "Please ensure that the outer instance has correct type and that the target class has 0-arg constructor."),
-                cause);
-    }
-
-    private static <T> InstantiationException noMatchingConstructor(Class<T> cls) {
-        return new InstantiationException(
-                join("Unable to create instance of '" + cls.getSimpleName() + "'.",
-                     "Unable to find a matching 1-arg constructor for the outer instance.")
-                , null);
+    private <T> InstantiationException paramsException(Class<T> cls, Exception e) {
+        return new InstantiationException(join(
+                "Unable to create instance of '" + cls.getSimpleName() + "'.",
+                "Please ensure that it has a constructor that matches these arguments: " + Arrays.toString(constructorArgs))
+                , e);
     }
 
     private static boolean paramsMatch(Class<?>[] types, Object[] params) {
```

Comment:
@ChristianSchwarz @szczepiq there's no new code here - this is the same code used originally for the enclosing type usecase, so although this is indeed an issue, it's not a new one, and thus I think it should be fixed in another patch (although probably still part of this PR.

The "strictest" solution, IMHO, would be to actually pass a list of arguments types - `useConstructor(Class[] types, Objcet[] args)`.

Alternatively, we can do something implicit:
- if an argument list matches a strictly more "specific" constructor, use it
- if you can't decide (e.g., the argument list is `{"String1", "String2"}`, and there are two constructors `MyClass(Object, String)` and `MyClass(String, Object)`, throw an exception.

What do you guys think?